src/libponyc/pass/docgen.c

```diff
@@ -1124,6 +1360,114 @@ static void doc_setup_dirs(docgen_t* docgen, ast_t* program, pass_opt_t* opt)
   doc_rm_star(docgen->sub_dir);
 }
 
+static void copy_file_content(FILE* source, FILE* dest)
+{
+  pony_assert(source != NULL && dest != NULL);
+  while(1)
+  {
+    int a = fgetc(source);
+    if(!feof(source))
+      fputc(a, dest);
+    else
+      break;
+  }
+}
+
+static void copy_doc_file(
+  const char* compiler_dir, 
+  const char* file_install_path, 
+  const char* file_source_path,
+  FILE* file_dest,
+  docgen_t* docgen
+)
+{
+  pony_assert(compiler_dir != NULL);
+  pony_assert(file_install_path != NULL);
+  pony_assert(file_source_path != NULL);
+  pony_assert(file_dest != NULL);
+  pony_assert(docgen != NULL);
+
+  // Find file in the compiler installation directory.
+  char file_path[FILENAME_MAX];
+  path_cat(compiler_dir, file_install_path, file_path);
+  FILE* file_source = fopen(file_path, "r");
+  if (file_source != NULL) {
+      copy_file_content(file_source, file_dest);
+      fclose(file_source);
+  } 
+  else {
+    // Find file in the source directory.
+    path_cat(compiler_dir, file_source_path, file_path);
+    file_source = fopen(file_path, "r");
+    if (file_source != NULL) {
+      copy_file_content(file_source, file_dest);
+      fclose(file_source);
+    }
+    else {
+      errorf(docgen->errors, NULL, "Cannot find either %s or %s." , file_install_path, file_source_path);
+    }
+  }
+  fclose(file_dest);
+}
+
+static void copy_doc_files(docgen_t* docgen)
```

Comment:
say we had a packaged pony theme for mkdocs, published on `PyPI` (this is how mkdocs themes are packaged and published http://www.mkdocs.org/user-guide/custom-themes/#packaging-themes ), we would not need to copy any of those special files. The only thing a user would need to do it so install both `mkdocs` and the imagined `ponylang-mkdocs-theme` when building the final docs, which is way less brittle than relying on another directory relative to the compiler exe (puts less burden on the packager of pony).

A custom theme can contain all the customized js, css and html-templates we'd want.

If we will not go this route, but copy the extra files from the docs-support dir, i'd suggest to have a whole theme in that dir and copy the whole dir over when generating docs. The `mkdocs.yml` would then use this option: http://www.mkdocs.org/user-guide/styling-your-docs/#using-the-theme-custom_dir . This way we have way more control over the theme and styling in general.